--- AvrCommon.am.old	2007-05-24 08:25:29.437500000 -0600
+++ AvrCommon.am	2007-05-24 08:28:59.984375000 -0600
@@ -34,9 +34,9 @@
 
 AM_CPPFLAGS = -I$(top_srcdir)/common -I$(top_srcdir)/include
 
-AVRLIB_CFLAGS = -gstabs -Wall -W -Wstrict-prototypes -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_CFLAGS)
+AVRLIB_CFLAGS = -gdwarf-2 -Wall -W -Wstrict-prototypes -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_CFLAGS)
 
-AVRLIB_ASFLAGS = -x assembler-with-cpp -Wa,-gstabs -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_ASFLAGS)
+AVRLIB_ASFLAGS = -x assembler-with-cpp -Wa,-gdwarf-2 -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_ASFLAGS)
 
 AM_CFLAGS = $(AVRLIB_CFLAGS)
 AM_CCASFLAGS = $(AM_CPPFLAGS) $(AVRLIB_ASFLAGS)
--- AvrCommonLibC.am.old	2007-06-03 13:53:46.021828600 -0600
+++ AvrCommonLibC.am	2007-06-03 13:54:04.807390200 -0600
@@ -37,7 +37,7 @@ AM_CPPFLAGS = -I$(top_srcdir)/common -I$
 
 AVRLIB_CFLAGS = -g -Wall -W -Wstrict-prototypes -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_CFLAGS)
 
-AVRLIB_ASFLAGS = -x assembler-with-cpp -Wa,-gstabs -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_ASFLAGS)
+AVRLIB_ASFLAGS = -x assembler-with-cpp -Wa,-gdwarf-2 -mmcu=$(AVR_TARGET) $(AVR_TARGET_DEFS) $(AVR_TARGET_ASFLAGS)
 
 AM_CFLAGS = $(AVRLIB_CFLAGS)
 AM_CCASFLAGS = $(AM_CPPFLAGS) $(AVRLIB_ASFLAGS)
